diff --git a/dom/external-libs/minizip-1.1/zip.c b/dom/external-libs/minizip-1.1/zip.c
index 3c34fc8..51f2f9d 100644
--- a/dom/external-libs/minizip-1.1/zip.c
+++ b/dom/external-libs/minizip-1.1/zip.c
@@ -1245,7 +1245,7 @@ extern int ZEXPORT zipOpenNewFileInZip4_64 (zipFile file, const char* filename,
         unsigned char bufHead[RAND_HEAD_LEN];
         unsigned int sizeHead;
         zi->ci.encrypt = 1;
-        zi->ci.pcrc_32_tab = get_crc_table();
+        zi->ci.pcrc_32_tab = (const unsigned long *)get_crc_table();
         /*init_keys(password,zi->ci.keys,zi->ci.pcrc_32_tab);*/
 
         sizeHead=crypthead(password,bufHead,RAND_HEAD_LEN,zi->ci.keys,zi->ci.pcrc_32_tab,crcForCrypting);
