diff --git a/src/osgPlugins/CMakeLists.txt b/src/osgPlugins/CMakeLists.txt
index 277dbebd5..a824dbb71 100644
--- a/src/osgPlugins/CMakeLists.txt
+++ b/src/osgPlugins/CMakeLists.txt
@@ -161,7 +161,7 @@ IF(NOT OSG_GLES1_AVAILABLE AND NOT OSG_GLES2_AVAILABLE)
     ENDIF()
 ENDIF()
 
-IF(COLLADA_FOUND AND OSG_CPP_EXCEPTIONS_AVAILABLE)
+IF(OSG_CPP_EXCEPTIONS_AVAILABLE)
     ADD_PLUGIN_DIRECTORY(dae)
 ENDIF()
 
